### DISCUSSION TITLE
How can I stop updating a style?

### DISCUSSION THREAD

**Discussion by Only-IceSoul** [software-mansion/react-native-reanimated#2854] [Q&A]
Being updated every time a value changes slows down the app too much in a list

It would be nice if it only updates when it is between a range
```
const animatedStyle = useAnimatedStylelistener( () => props.shareValue.value , (v)=>{

return {
        opacity: v / 100
}

},  0,100 )


```

Actually I don't know what slows down the application, but when I return an empty style {} it improves performance despite doing calculations inside

**Comment by JasonCareter** [comment]
where the style listener come from ? you'd better use the hook useAnimatedStyle . by the way , if you wanna restrict the animation value to a range, you can use **interpolate** exported from reanimated.

**Comment by Only-IceSoul** [comment]
style listener is what the aninimatedStyle  should be . The interpolation is too slow, it is preferable to make a condition. 
Whatever. most people use ios
